src/units.cpp

```diff
@@ -181,12 +181,17 @@ bool updateUnitMultiplier(const UnitsPtr &units, int direction, double &multipli
         double expMult;
         double standardMult = 0.0;
         double prefixMult = 0.0;
+        int prefixInt;
         for (size_t i = 0; i < units->unitCount(); ++i) {
             units->unitAttributes(i, ref, pre, exp, expMult, id);
             mult = std::log10(expMult);
 
             if (isStandardPrefixName(pre)) {
                 prefixMult = standardPrefixList.at(pre);
+            } else if (pre.empty()) {
+                prefixMult = 0;
+            } else if (convertToInt(pre, prefixInt)) {
+                prefixMult = prefixInt;
             } else {
                 return false;
             }
```

Comment:
I would use our new `convertPrefixToInt()` method for this, which means that `convertPrefixToInt()` would have to be reworked a bit to return `true` or `false` depending on whether the conversion was successful or not. IOW, I don't like "duplication" of code.

Reply:
9a0c7b9